# Phase 1 Project

**Authors:** Jonathan Holt
***

## Overview

A one-paragraph overview of the project, including the business problem, data, methods, results and recommendations.

## Business Problem

Microsoft sees all the big companies creating original video content and they want to get in on the fun. They have decided to create a new movie studio, but they don’t know anything about creating movies. You are charged with exploring what types of films are currently doing the best at the box office. You must then translate those findings into actionable insights that the head of Microsoft's new movie studio can use to help decide what type of films to create.
***
Questions to consider:
* What are the business's pain points related to this project?
* How did you pick the data analysis question(s) that you did?
* Why are these questions important from a business perspective?
***

## Data Understanding

Describe the data being used for this project.
***
Questions to consider:
* Where did the data come from, and how do they relate to the data analysis questions?
* What do the data represent? Who is in the sample and what variables are included?
* What is the target variable?
* What are the properties of the variables you intend to use?
***

## Questions that I want to answer:
1. **What movies are doing well at the box office?**
2. **What genres should Microsoft make?**
3. **How much should Microsoft spend on these movies?**
4.


## Data Preparation


In [ ]:
#Describe and justify the process for preparing the data for analysis.

#Questions to consider:

#Were there variables you dropped or created?
#How did you address missing values or outliers?
#Why are these choices appropriate given the data and the business problem?


### START HERE!

### Importing

In [1]:
# Import standard packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('seaborn') #style that I am using for my visualizations
import seaborn as sns

%matplotlib inline

In [2]:
# Here you run your code to explore the data
import glob, os
fpath = 'zippedData/'
os.listdir(fpath)

['imdb.title.crew.csv.gz',
 'tmdb.movies.csv.gz',
 'imdb.title.akas.csv.gz',
 'imdb.title.ratings.csv.gz',
 'imdb.name.basics.csv.gz',
 'rt.reviews.tsv.gz',
 'imdb.title.basics.csv.gz',
 'rt.movie_info.tsv.gz',
 'tn.movie_budgets.csv.gz',
 'bom.movie_gross.csv.gz',
 'imdb.title.principals.csv.gz']

In [3]:
query = fpath+"*.gz"

file_list=glob.glob(query)
file_list

['zippedData/imdb.title.crew.csv.gz',
 'zippedData/tmdb.movies.csv.gz',
 'zippedData/imdb.title.akas.csv.gz',
 'zippedData/imdb.title.ratings.csv.gz',
 'zippedData/imdb.name.basics.csv.gz',
 'zippedData/rt.reviews.tsv.gz',
 'zippedData/imdb.title.basics.csv.gz',
 'zippedData/rt.movie_info.tsv.gz',
 'zippedData/tn.movie_budgets.csv.gz',
 'zippedData/bom.movie_gross.csv.gz',
 'zippedData/imdb.title.principals.csv.gz']

In [4]:
tables = {}

for file in file_list:
    print('---'*20)
    file_name = file.replace('zippedData/', '').replace('.', '_')
    print(file_name)
    
    
    
    if 'tsv.gz' in file:
        temp_df = pd.read_csv(file, sep= "\t", encoding = "latin-1")
    else:
        temp_df = pd.read_csv(file)
    
    display(temp_df.head(), temp_df.tail())
    tables[file_name] = temp_df

------------------------------------------------------------
imdb_title_crew_csv_gz


tconst                      directors              writers
0  tt0285252                      nm0899854            nm0899854
1  tt0438973                            NaN  nm0175726,nm1802864
2  tt0462036                      nm1940585            nm1940585
3  tt0835418                      nm0151540  nm0310087,nm0841532
4  tt0878654  nm0089502,nm2291498,nm2292011            nm0284943

tconst              directors     writers
146139  tt8999974             nm10122357  nm10122357
146140  tt9001390              nm6711477   nm6711477
146141  tt9001494  nm10123242,nm10123248         NaN
146142  tt9004986              nm4993825   nm4993825
146143  tt9010172                    NaN   nm8352242

------------------------------------------------------------
tmdb_movies_csv_gz


Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

Unnamed: 0        genre_ids      id original_language  \
26512       26512         [27, 18]  488143                en   
26513       26513         [18, 53]  485975                en   
26514       26514     [14, 28, 12]  381231                en   
26515       26515  [10751, 12, 28]  366854                en   
26516       26516         [53, 27]  309885                en   

              original_title  popularity release_date                  title  \
26512  Laboratory Conditions         0.6   2018-10-13  Laboratory Conditions   
26513        _EXHIBIT_84xxx_         0.6   2018-05-01        _EXHIBIT_84xxx_   
26514           The Last One         0.6   2018-10-01           The Last One   
26515           Trailer Made         0.6   2018-06-22           Trailer Made   
26516             The Church         0.6   2018-10-05             The Church   

       vote_average  vote_count  
26512           0.0           1  
26513           0.0           1  
26514           0.0           1  
26515           0.0           1  
26516           0.0           1

------------------------------------------------------------
imdb_title_akas_csv_gz


title_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   

  language        types   attributes  is_original_title  
0       bg          NaN          NaN                0.0  
1      NaN  imdbDisplay          NaN                0.0  
2      NaN  imdbDisplay          NaN                0.0  
3      NaN          NaN  short title                0.0  
4      NaN  imdbDisplay          NaN                0.0

title_id  ordering               title region language        types  \
331698  tt9827784         2  Sayonara kuchibiru    NaN      NaN     original   
331699  tt9827784         3       Farewell Song    XWW       en  imdbDisplay   
331700  tt9880178         1         La atención    NaN      NaN     original   
331701  tt9880178         2         La atención     ES      NaN          NaN   
331702  tt9880178         3       The Attention    XWW       en  imdbDisplay   

       attributes  is_original_title  
331698        NaN                1.0  
331699        NaN                0.0  
331700        NaN                1.0  
331701        NaN                0.0  
331702        NaN                0.0

------------------------------------------------------------
imdb_title_ratings_csv_gz


tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

tconst  averagerating  numvotes
73851  tt9805820            8.1        25
73852  tt9844256            7.5        24
73853  tt9851050            4.7        14
73854  tt9886934            7.0         5
73855  tt9894098            6.3       128

------------------------------------------------------------
imdb_name_basics_csv_gz


nconst       primary_name  birth_year  death_year  \
0  nm0061671  Mary Ellen Bauder         NaN         NaN   
1  nm0061865       Joseph Bauer         NaN         NaN   
2  nm0062070         Bruce Baum         NaN         NaN   
3  nm0062195       Axel Baumann         NaN         NaN   
4  nm0062798        Pete Baxter         NaN         NaN   

                                 primary_profession  \
0         miscellaneous,production_manager,producer   
1        composer,music_department,sound_department   
2                        miscellaneous,actor,writer   
3  camera_department,cinematographer,art_department   
4  production_designer,art_department,set_decorator   

                          known_for_titles  
0  tt0837562,tt2398241,tt0844471,tt0118553  
1  tt0896534,tt6791238,tt0287072,tt1682940  
2  tt1470654,tt0363631,tt0104030,tt0102898  
3  tt0114371,tt2004304,tt1618448,tt1224387  
4  tt0452644,tt0452692,tt3458030,tt2178256

nconst         primary_name  birth_year  death_year  \
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

           primary_profession     known_for_titles  
606643                actress                  NaN  
606644                actress  tt9090932,tt8737130  
606645                actress  tt8734436,tt9615610  
606646               producer                  NaN  
606647  director,actor,writer            tt8743182

------------------------------------------------------------
rt_reviews_tsv_gz


id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

id                                             review rating   fresh  \
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic        publisher                date  
54427      Laura Sinagra           1    Village Voice  September 24, 2002  
54428  Michael Szymanski           0       Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0  EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0   Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0     Showbizz.net   November 12, 2002

------------------------------------------------------------
imdb_title_basics_csv_gz


tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

tconst                                primary_title  \
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes       genres  
146139            123.0        Drama  
146140              NaN  Documentary  
146141              NaN       Comedy  
146142            116.0          NaN  
146143              NaN  Documentary

------------------------------------------------------------
rt_movie_info_tsv_gz


id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

id                                           synopsis rating  \
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures

------------------------------------------------------------
tn_movie_budgets_csv_gz


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

id  release_date                          movie production_budget  \
5777  78  Dec 31, 2018                         Red 11            $7,000   
5778  79   Apr 2, 1999                      Following            $6,000   
5779  80  Jul 13, 2005  Return to the Land of Wonders            $5,000   
5780  81  Sep 29, 2015           A Plague So Pleasant            $1,400   
5781  82   Aug 5, 2005              My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041

------------------------------------------------------------
bom_movie_gross_csv_gz


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

------------------------------------------------------------
imdb_title_principals_csv_gz


tconst  ordering     nconst  category       job        characters
0  tt0111414         1  nm0246005     actor       NaN       ["The Man"]
1  tt0111414         2  nm0398271  director       NaN               NaN
2  tt0111414         3  nm3739909  producer  producer               NaN
3  tt0323808        10  nm0059247    editor       NaN               NaN
4  tt0323808         1  nm3579312   actress       NaN  ["Beth Boothby"]

tconst  ordering      nconst  category       job  \
1028181  tt9692684         1   nm0186469     actor       NaN   
1028182  tt9692684         2   nm4929530      self       NaN   
1028183  tt9692684         3  nm10441594  director       NaN   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                   NaN  
1028184                   NaN  
1028185                   NaN

### Formating Cell
I am keeping all of my display commands here so I can easily find them if/when I need
to change anything.

In [5]:
pd.set_option('display.max_rows', 1000)
#change the amount of rows displayed
pd.options.display.float_format = '{:,.0f}'.format 
#This takes the decimal places out of floats. I may need to change this for some of the other features.

In [7]:
#sorting by worldwide gross
tables['tn_movie_budgets_csv_gz'].sort_values(by='worldwide_gross', ascending=False).head(5)

id  release_date                   movie production_budget  \
3737  38  Aug 21, 2009  Fifty Dead Men Walking       $10,000,000   
3432  33  Sep 30, 2005                    Duma       $12,000,000   
5062  63   Apr 1, 2011               Insidious        $1,500,000   
883   84   Apr 2, 2004                 Hellboy       $60,000,000   
5613  14  Mar 21, 1980                 Mad Max          $200,000   

     domestic_gross worldwide_gross  
3737             $0        $997,921  
3432       $870,067        $994,790  
5062    $54,009,150     $99,870,886  
883     $59,623,958     $99,823,958  
5613     $8,750,000     $99,750,000

Something is wrong here. I assume that it is sorting anything with a '9' in the front first.
I presume that the dollar amounts are strings instead of integers. I should also check for
null values while I'm checking for this.

### Table Aliases
Setting an alias for each table so that it is easier to use them.

In [8]:
table1 = tables['imdb_title_crew_csv_gz']
table2 = tables['tmdb_movies_csv_gz']
table3 = tables['imdb_title_akas_csv_gz']
table4 = tables['imdb_title_ratings_csv_gz']
table5 = tables['imdb_name_basics_csv_gz']
table6 = tables['rt_reviews_tsv_gz']
table7 = tables['imdb_title_basics_csv_gz']
table8 = tables['rt_movie_info_tsv_gz']
table9 = tables['tn_movie_budgets_csv_gz']
table10 = tables['bom_movie_gross_csv_gz']
table11 = tables['imdb_title_principals_csv_gz']

In [9]:
table9.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


## Initial Data Cleaning

Sure enough, everything is a string except for the ID field. I will convert all of the financial information
to floats and while I'm at it. I will also convert the release date field to date/time.

In [10]:
table9.isna().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

There are no null values, but there are likely placeholders

I will now clean the three financial columns. Removing the $, removing the comma, then converting to float.

In [11]:
cleaned_budget = table9['production_budget'].map(lambda x: x.replace('$',' '))
cleaned_budget = cleaned_budget.map(lambda x: x.replace(',',''))
cleaned_budget = cleaned_budget.astype(float)

cleaned_domestic = table9['domestic_gross'].map(lambda x: x.replace('$',' '))
cleaned_domestic =cleaned_domestic.map(lambda x: x.replace(',',''))
cleaned_domestic = cleaned_domestic.astype(float)

cleaned_worldwide = table9['worldwide_gross'].map(lambda x: x.replace('$',' '))
cleaned_worldwide = cleaned_worldwide.map(lambda x: x.replace(',',''))
cleaned_worldwide = cleaned_worldwide.astype(float)

In [13]:
#Putting my cleaned data into the table.
table9["production_budget"] = cleaned_budget
table9["domestic_gross"] = cleaned_domestic
table9["worldwide_gross"] = cleaned_worldwide
table9 = table9.sort_values(by='worldwide_gross', ascending=False)

#converting release_date to datetime format.
cleaned_release_date = pd.to_datetime(table9['release_date'])
table9['release_date']= cleaned_release_date

#Creating a column 'release_year' to more easily search general release dates, etc.
table9['release_year'] = table9['release_date'].dt.year

table9.head(5)


id release_date                                 movie  production_budget  \
0    1   2009-12-18                                Avatar        425,000,000   
42  43   1997-12-19                               Titanic        200,000,000   
5    6   2015-12-18  Star Wars Ep. VII: The Force Awakens        306,000,000   
6    7   2018-04-27                Avengers: Infinity War        300,000,000   
33  34   2015-06-12                        Jurassic World        215,000,000   

    domestic_gross  worldwide_gross  release_year  
0      760,507,625    2,776,345,279          2009  
42     659,363,944    2,208,208,395          1997  
5      936,662,225    2,053,311,220          2015  
6      678,815,482    2,048,134,200          2018  
33     652,270,625    1,648,854,864          2015

In [ ]:
#converting release_date to datetime format.
#cleaned_release_date = pd.to_datetime(table9['release_date'])
#table9['release_date']= cleaned_release_date
#table9.head(1)

**What is the range of release dates in this data set?**

In [14]:
table9.sort_values(by='release_date', ascending=False).tail(3)

id release_date                         movie  production_budget  \
5614  15   1916-12-24  20,000 Leagues Under the Sea            200,000   
5523  24   1916-09-05                   Intolerance            385,907   
5677  78   1915-02-08         The Birth of a Nation            110,000   

      domestic_gross  worldwide_gross  release_year  
5614       8,000,000        8,000,000          1916  
5523               0                0          1916  
5677      10,000,000       11,000,000          1915

In [15]:
table9.sort_values(by='release_date', ascending=False).head(3)

id release_date                   movie  production_budget  \
194   95   2020-12-31                Moonfall        150,000,000   
1205   6   2020-12-31  Hannibal the Conqueror         50,000,000   
535   36   2020-02-21        Call of the Wild         82,000,000   

      domestic_gross  worldwide_gross  release_year  
194                0                0          2020  
1205               0                0          2020  
535                0                0          2020

The earliest movie is from 1915. The most recent are from 2020.
- I doubt that I will need data from movies from more than a few decades ago. I will slice out recent movie data as I proceed with my data cleaning.
- The movies from 2020 have no performance data. It will be okay to exclude them from analysis as they aren't relevant.
- The bulk of the data starts in 1996 where there are 100+ movies every year afterward until 2018. There is a little bit of info for 2019.

### Calculating Total Profit and ROI
- The first way that I will be manipulating the cleaned data is to calculate two financial performance metrics: Total Profit, and Return on Investment (ROI)

In [16]:
#creating a column that calculates the profit for each movie. (gross - budget)
total_profit = table9.apply(lambda x: x['worldwide_gross'] - x['production_budget'], axis=1)
table9['total_profit'] = total_profit

#creating a column that calculates ROI for each movie. (Profit divided by budget)
roi = table9.apply(lambda x: x['total_profit'] / x['production_budget'], axis=1)
table9['ROI'] = roi

table9


id release_date                                 movie  \
0      1   2009-12-18                                Avatar   
42    43   1997-12-19                               Titanic   
5      6   2015-12-18  Star Wars Ep. VII: The Force Awakens   
6      7   2018-04-27                Avengers: Infinity War   
33    34   2015-06-12                        Jurassic World   
...   ..          ...                                   ...   
4856  57   2014-11-14                      Always Woodstock   
4980  81   1971-04-28                               Bananas   
4850  51   2014-12-31                     Not Safe For Work   
4983  84   2007-07-07                              Rockaway   
4068  69   2008-12-12                The Kings of Appletown   

      production_budget  domestic_gross  worldwide_gross  release_year  \
0           425,000,000     760,507,625    2,776,345,279          2009   
42          200,000,000     659,363,944    2,208,208,395          1997   
5           306,000,000     936,662,225    2,053,311,220          2015   
6           300,000,000     678,815,482    2,048,134,200          2018   
33          215,000,000     652,270,625    1,648,854,864          2015   
...                 ...             ...              ...           ...   
4856          2,400,000               0                0          2014   
4980          2,000,000               0                0          1971   
4850          2,500,000               0                0          2014   
4983          2,000,000               0                0          2007   
4068          7,000,000               0                0          2008   

      total_profit  ROI  
0    2,351,345,279    6  
42   2,008,208,395   10  
5    1,747,311,220    6  
6    1,748,134,200    6  
33   1,433,854,864    7  
...            ...  ...  
4856    -2,400,000   -1  
4980    -2,000,000   -1  
4850    -2,500,000   -1  
4983    -2,000,000   -1  
4068    -7,000,000   -1  

[5782 rows x 9 columns]

In [17]:
#A quick snapshot of the information. 
table9.describe()

id  production_budget  domestic_gross  worldwide_gross  release_year  \
count 5,782              5,782           5,782            5,782         5,782   
mean     50         31,587,757      41,873,327       91,487,461         2,004   
std      29         41,812,077      68,240,597      174,719,969            13   
min       1              1,100               0                0         1,915   
25%      25          5,000,000       1,429,534        4,125,415         2,000   
50%      50         17,000,000      17,225,945       27,984,448         2,007   
75%      75         40,000,000      52,348,662       97,645,836         2,012   
max     100        425,000,000     936,662,225    2,776,345,279         2,020   

       total_profit   ROI  
count         5,782 5,782  
mean     59,899,704     4  
std     146,088,881    30  
min    -200,237,650    -1  
25%      -2,189,071    -1  
50%       8,550,286     1  
75%      60,968,502     3  
max   2,351,345,279 1,799

## Getting Genre Data
I am happy with where my financial data is at the moment. Now I want to find a way to indentify the **genres** of these movies as that will help me answer the specific question of what ***types*** **of movies should be made.**

In [ ]:
#After this, I plan on investigating review scores, giving me 
#**financials, review scores, and genres to work with.**

In [18]:
table7.head(2)

tconst                    primary_title   original_title  start_year  \
0  tt0063540                        Sunghursh        Sunghursh        2013   
1  tt0066787  One Day Before the Rainy Season  Ashad Ka Ek Din        2019   

   runtime_minutes              genres  
0              175  Action,Crime,Drama  
1              114     Biography,Drama

IMDB Basics (Table 7) seems to be the best place to get my genres.

In [19]:
table7.sort_values(by='start_year', ascending=True).head(3)


tconst                                      primary_title  \
9599   tt1566491                            Brainiacs in La La Land   
43264  tt2578092  Fireplace for your Home: Crackling Fireplace w...   
11550  tt1634300                                          Role/Play   

                                          original_title  start_year  \
9599                             Brainiacs in La La Land        2010   
43264  Fireplace for your Home: Crackling Fireplace w...        2010   
11550                                          Role/Play        2010   

       runtime_minutes         genres  
9599               nan         Comedy  
43264               61          Music  
11550               85  Drama,Romance

IMDB Basics (Table 7) seems to be the best place to get my genres.
- This data only goes back to 2010. This will still be useful as Microsoft will want current information on
which genres are popular.
- I want to see what other data I can collect from the IMDB datasets before attempting to merge the titles from this dataset with my financial data.
- Many of the IMDB datasets contain a field 'tconst' which will link more data to the genres and titles that I have here.

In [21]:
#merging, using the tconst as a key.
imdb_df = pd.merge(table7, table4, left_on= 'tconst', right_on= 'tconst', how='left')
imdb_df= pd.merge(imdb_df, table1, left_on= 'tconst', right_on= 'tconst', how='left')
imdb_df= pd.merge(imdb_df, table11, left_on= 'tconst', right_on= 'tconst', how='left')
imdb_df.head(3)

tconst primary_title original_title  start_year  runtime_minutes  \
0  tt0063540     Sunghursh      Sunghursh        2013              175   
1  tt0063540     Sunghursh      Sunghursh        2013              175   
2  tt0063540     Sunghursh      Sunghursh        2013              175   

               genres  averagerating  numvotes  directors  \
0  Action,Crime,Drama              7        77  nm0712540   
1  Action,Crime,Drama              7        77  nm0712540   
2  Action,Crime,Drama              7        77  nm0712540   

                                   writers  ordering     nconst  category  \
0  nm0023551,nm1194313,nm0347899,nm1391276        10  nm0006210  composer   
1  nm0023551,nm1194313,nm0347899,nm1391276         1  nm0474801     actor   
2  nm0023551,nm1194313,nm0347899,nm1391276         2  nm0904537   actress   

   job                       characters  
0  NaN                              NaN  
1  NaN  ["Kundan S. Prasad","Bajrangi"]  
2  NaN      ["Munni","Laila-E-Aasmaan"]

There are a lot of duplicates due to the personnel attached to each movie getting their own record. I doubt that I will need to check the personnel, so let's drop the duplicates.

In [23]:
imdb_df = imdb_df.drop_duplicates('original_title', keep='first')
imdb_df.head(3)

tconst                    primary_title              original_title  \
0   tt0063540                        Sunghursh                   Sunghursh   
10  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
17  tt0069049       The Other Side of the Wind  The Other Side of the Wind   

    start_year  runtime_minutes              genres  averagerating  numvotes  \
0         2013              175  Action,Crime,Drama              7        77   
10        2019              114     Biography,Drama              7        43   
17        2018              122               Drama              7     4,517   

    directors                                  writers  ordering     nconst  \
0   nm0712540  nm0023551,nm1194313,nm0347899,nm1391276        10  nm0006210   
10  nm0002411                                      NaN         1  nm0045119   
17  nm0000080                      nm0000080,nm0462648        10  nm0613657   

    category  job characters  
0   composer  NaN        NaN  
10   actress  NaN        NaN  
17    editor  NaN        NaN

I am dropping a lot of the other fields as well. If I need them again, I can re-merge them using the tconst. I am dropping: *directors, writers, ordering, nconst, category, job, characters*.

In [ ]:
imdb_df = imdb_df.drop(['directors', 'writers', 'ordering', 'nconst', 'category', 'job', 'characters'], axis = 1)
imdb_df.head(2)

### Merging the Dataframes

I want to now merge this imdb_df with my financial information from table9. I know that if I just do a merge on primary title, some of the results will be incorrect. (Avatar is the top grossing movie but is listed as a horror movie)


In [ ]:
#inner join with the indiciator column so I can track where everything is coming from.
merged_originals = pd.merge(table9, imdb_df, left_on= 'movie', right_on= 'original_title', how='left', indicator=True)
merged_originals.head()

In [ ]:
merged_originals.info()

 5782 records after the join.

I am going ot slice into two datasets. One that merged on original title, and everything else that still needs to merge.

In [ ]:
correct_merge_df = merged_originals.loc[(merged_originals["_merge"] == "both")]
need_to_merge = merged_originals.loc[(merged_originals["_merge"] != "both")]


In [ ]:
#need_to_merge.head(3)

In [ ]:
#correct_merge_df = correct_merge_df.drop(['_merge'], axis=1) #dropping '_merge' so I can continue

Locating Avatar and moving it to the correct_merge_df as I can tell from the release_year, and original_title that the horror movie Avatar is not the one that was the best grossing movie of all time.

In [ ]:
#avatar = merged_originals.loc[(merged_originals['movie'] == 'Avatar')]
#correct_merge_df = correct_merge_df.append(avatar, ignore_index = True)
correct_merge_df = correct_merge_df.drop(0)

In [ ]:
correct_merge_df.sort_values('total_profit', ascending=False).head(3)

In [ ]:
#need to drop the indicator column if I want to use it again.
need_to_merge = need_to_merge.drop(['_merge'], axis=1)

In [ ]:
#dropping Avatar as I know that it is not merging correctly.
need_to_merge = need_to_merge.drop(0)
need_to_merge.head(3)

In [ ]:
#now merging need_to_merge  with primary name.
merged_with_primary = pd.merge(need_to_merge, imdb_df, left_on= 'movie', right_on= 'primary_title', how='left', indicator=True)
merged_with_primary.head()

In [ ]:
merged_with_primary = merged_with_primary.drop(merged_with_primary.columns[9:16], axis = 1)
merged_with_primary.head(3)

In [ ]:
merged_with_primary = merged_with_primary.drop(merged_with_primary.columns[9], axis = 1)
merged_with_primary.head(3)

In [ ]:
#dropping indicator column
merged_with_primary = merged_with_primary.drop(merged_with_primary.columns[16], axis = 1)
merged_with_primary.head(3)

In [ ]:
#Renaming the columns which contain _y
merged_with_primary.rename(columns = {'averagerating_y':'averagerating', 'numvotes_y':'numvotes',
                              'primary_title_y':'primary_title', 'original_title_y':'original_title',
                              'start_year_y': 'start_year', 'runtime_minutes_y': 'runtime_minutes',
                              'genres_y' : 'genres'}, inplace = True)

In [ ]:
merged_with_primary.head(3)

Appending "merged with primary" to correct merge to make project_df.

In [ ]:
project_df =  correct_merge_df.append(merged_with_primary, ignore_index=True)
project_df.head()

In [ ]:
#dropping the columns that I missed before.
project_df = project_df.drop(project_df[['_merge', 'tconst_y']], axis = 1)
project_df

In [ ]:
project_df.info()

In [ ]:
#checking for duplicates
duplicates = project_df[project_df.duplicated()]
print(len(duplicates))
duplicates.head()

In [ ]:
#storing this df as project_df_master to keep all of this data safe before I start slicing out what I need.
project_df_master = project_df

### Cleaning the genre data

In [ ]:
project_df[['genre_1', 'genre_2', 'genre_3']] = project_df['genres'].str.split(pat=",", expand=True)
project_df.head(3)

In [ ]:
#adding my new genre columns to the master df
project_df_master = project_df

In [ ]:
project_df_master.head(3)

Slicing out the columns that I want to use for my analysis.

In [ ]:
project_analysis_df = project_df_master[['movie','release_year', 'production_budget', 'total_profit', 'ROI',
                                       'averagerating', 'numvotes','runtime_minutes', 'genre_1', 'genre_2',
                                        'genre_3', ]]
project_analysis_df.head(3)

## Data Analysis

### Analysis of the Project Dataframe

In [ ]:
project_analysis_df.info()

In [ ]:
project_analysis_df.describe()

**Project Analysis dataframe:**

- Contains financial data from The Numbers and genre and ratings from IMDB.
- 5796 movies in this dataset
- 2250 have ratings and genre information
- Earliest movie is 1915.
- Most recent movie is 2020.

**Next Steps:**
- Recent data will be the most useful, so I will slice out the last 30-40 years of data.
- It would also be beneficial to look at the data in 10 year slices to analyze trends.
- Also remove movies that are too new to have relevent data. (ex. 2020 movies that have a budget but no revenue)

### Analysis of Recent Movies (2000-2019)

In [ ]:
#creating recent_movies which contains all movies since 2000.
recent_movies = project_analysis_df
recent_movies = recent_movies[recent_movies['release_year'] >= 2000]
recent_movies = recent_movies[recent_movies['release_year'] < 2020]
recent_movies.info()

In [ ]:
recent_movies.describe()

**Recent Movies (2000 - 2019):**
- There are 4,395 movies, which is a good sample size.
- Nearly half of them have genre information.
- A movie needs to be near the median to make a profit.
- The median and the mean are very different. I need to look for outliers.
- The better performing movies have: slightly higher averagerating and longer average runtime


### Insert Genre Analysis HERE!

### Analysis of the last 4 decades

I am splitting the dataset. Most of my genre information is for 2010-2019, so I want to isolate it. It will also be useful to look at the data in 10(ish) year increments so see which trends change, and which stay consistent. I also plan on getting data for 80s and 90s in a little while. 

In [ ]:
recent_movies_2010 = recent_movies[recent_movies['release_year'] >= 2010]
recent_movies_2000 = recent_movies[recent_movies['release_year'] < 2010]

In [ ]:
recent_movies_2010.info()

In [ ]:
recent_movies_2000.info()

Roughly 2200 movies in each dataframe after the split.

In [ ]:
recent_movies_2010.describe()

**For 2010 - 2019 releases. On average:**
- Need to be in the top 50% to get a return on investement. 
- If in the top 50%, the median budget is 16 million, with a profit of 8.5 million.
- The IMDB ratings all see to be middle of the road, although the movies in the top quartile have a higher average (barely)
- I have genre information for most movies in this dataset, so it is what I will use for all genre analysis.
- This is fine because Microsoft will want the most recent data about what movies people want to see.

In [ ]:
recent_movies_2000.describe()

### attempting vizualizations

In [ ]:
#FIXES FORMATING FOR DOLLAR AMOUNTS IN VISUALIZATIONS
#USE ax.yaxis.set_major_formatter(price_fmt_mill)
from matplotlib.ticker import FuncFormatter

def millions(x,pos):
    """function for use wth matplotlib FuncFormatter -  formats money in millions"""
    return f"${x*1e-6:,}M"

# Create the formatter
price_fmt_mill =FuncFormatter(millions)


In [ ]:
#attempting a visualization
sns.barplot(data=recent_movies_2010, x="production_budget", y="total_profit");

In [ ]:
#creating a variable for all movies 1980-2019
recent_movies_2 = project_analysis_df
recent_movies_2 = recent_movies_2[recent_movies_2['release_year'] >= 1990]
recent_movies_2 = recent_movies_2[recent_movies_2['release_year'] < 2019]
recent_movies_2.sort_values('release_year', ascending=True).head(3)

In [ ]:
recent_movies_2.sort_values('production_budget').tail(1000)

In [ ]:
#attempting to show production budget, total profit, and ROI over time.
sns.lineplot(x="release_year", y="production_budget", data=recent_movies_2, label="Production Budget")
sns.lineplot(x="release_year", y="total_profit", data=recent_movies_2, label="Total Profit")

plt.show();

In [ ]:
df = recent_movies_2

fig, ax = plt.subplots(figsize = (20,10))

ax.plot(df['release_year'],y = df['production_budget'])

ax.set_xlabel('Release Year')
ax.set_ylabel('Production Budget')
ax.set_title('Insert Title Here');


In [ ]:
#another attempt
sns.lineplot(x="release_year", y="total_profit", data=recent_movies_2)
plt.show()

In [ ]:
#another attempt
sns.lineplot(x="release_year", y="ROI", data=recent_movies_2)
plt.show()

**For 2000 - 2009 releases. On average:**
- Movies need to be in the top 50% to get a return on investement. (same as 2010-2019)
- If in the top 50%, the median budget is 20 million, with a profit of 7 million.
- Median budget seems to have gone down slightly in 2010-2019 - I will look for outliers.
- The IMDB ratings all see to be middle of the road, although the movies in the top quartile have a higher average. (barely)
- Data is largely the same as the more recent dataset (2010-2019)


In [ ]:
#Attempting to plot my findings from the two recent movie datasets. 

df = recent_movies_2

# Set up plot
fig, ax = plt.subplots(figsize=(20,10))

# release year vs. profit
ax.bar(df["release_year"], df["total_profit"], color="purple")
ax.set_title("total profit by year", size=20)
ax.yaxis.set_major_formatter(price_fmt_mill)
ax.set_xlabel("Release Year", size=20)
ax.set_ylabel("Total Profit", size=20);

In [ ]:
df = recent_movies_2

fig, ax = plt.subplots(figsize = (20,10))

ax.scatter(df['production_budget'], df['total_profit'])


ax.set_xlabel('Production Budget')
ax.set_ylabel('Total Profit (in $millions)')
ax.set_title('correlation between budget and profit')
ax.yaxis.set_major_formatter(price_fmt_mill)
plt.tight_layout()

In [ ]:
df = recent_movies_2

fig, ax = plt.subplots(figsize = (20,10))

ax.scatter(df['production_budget'], df['ROI'])


ax.set_xlabel('Production Budget')
ax.set_ylabel('ROI')
ax.set_title('correlation between budget and ROI')
plt.tight_layout()

### Visualizations for 2010-2019

In [ ]:
recent_movies_2010.head(3)

In [ ]:
sns.lineplot(x="release_year", y="ROI", data=recent_movies_2010)
plt.show()

In [ ]:
sns.lineplot(x="release_year", y="production_budget", data=recent_movies_2010, label="Production Budget")
sns.lineplot(x="release_year", y="total_profit", data=recent_movies_2010, label="Total Profit")
plt.show();

In [ ]:
df = recent_movies_2010

# Set up plot
fig, ax = plt.subplots(figsize=(20,10))

# release year vs. profit
ax.bar(df["release_year"], df["total_profit"], color="purple")
ax.set_title("total profit by year", size=20)
ax.yaxis.set_major_formatter(price_fmt_mill)
ax.set_xlabel("Release Year", size=20)
ax.set_ylabel("Total Profit", size=20);

### VISUALIZATIONS FOR TOP MOVIES

In [ ]:
df = top_profit

# Set up plot
fig, ax = plt.subplots(figsize=(20,10))

# release year vs. profit
ax.bar(df["release_year"], df["total_profit"], color="green")
ax.set_title("total profit by year", size=20)
ax.yaxis.set_major_formatter(price_fmt_mill)
ax.set_xlabel("Release Year", size=20)
ax.set_ylabel("Total Profit", size=20);

### Analysis of the last 4 decades

In [ ]:
movies_90s = project_analysis_df[project_analysis_df['release_year'] <= 1999]
movies_90s = movies_90s[movies_90s['release_year'] >= 1990]
movies_90s.describe()

**For 90s releases. On average:**
- Need to be in the top 50% to get a return on investement, however it is easier to break even. (first quartile)
- If in the top 50%, the median budget is 22.5 million, with a profit of 9 million.
- the IMDB ratings all see to be middle of the road, although the movies in the top quartile have a higher average. (barely)
- data is largely the same as the more recent data

In [ ]:
movies_80s = project_analysis_df[project_analysis_df['release_year'] <= 1989]
movies_80s = movies_80s[movies_80s['release_year'] >= 1980]
movies_80s.describe()

**For 80s releases. On average:**
- Need to be in the top 50% to get a return on investement, however it is easier to break even. (first quartile)
- If in the top 50%, the median budget is 12 million, with a profit of 15.5 million.
- the IMDB ratings all see to be middle of the road, although the movies in the top quartile have a higher average. (barely)
- data is largely the same as the more recent data

### Profit Analysis

**How do we determine what movies are the most successful?**
- Using the 2010-2019 dataset, I will slice out the **most successful** movies.
- The two metrics that I will be using are **total profit and return on investment (ROI)**
- I will then slice out the top 500 movies as that is roughly the top quartile for this dataset. 
- These movies would be good to emulate as we want to be as successful as possible.

In [ ]:
#arranging the 2010 - 2019 movies by most profit and most ROI.
#profit_df= recent_movies_2010.sort_values('total_profit', ascending=False)
#roi_df = recent_movies_2010.sort_values('ROI', ascending=False)
#top_profit = profit_df[0:500]
#all_profit = profit_df[0:1250] #new variable as of 9/30
#top_roi = roi_df[0:500]

In [ ]:
#movies that did not make a profit
#loss_df= recent_movies_2010.sort_values('total_profit', ascending=True)
#loss_df = loss_df[0:800]

In [ ]:
#trying to make profit df apples to apples with loss df
#profit_df_2 = recent_movies_2010.sort_values('total_profit', ascending=False)
#profit_df_2 = profit_df_2[0:1000]

In [ ]:
#loss_genre_df = genre_df(loss_df)
#loss_genre_df = loss_genre_df.reset_index()
#loss_genre_df = loss_genre_df.drop(loss_genre_df[['level_0','genre_1', 'genre_2', 'genre_3']], axis=1)
#loss_genre_df = loss_genre_df.sort_values('total', ascending=False)
#loss_genre_df

**INSERT VISUALIZATION OF PROFIT AND ROI**

### Preparing for Genre Analysis 

**What genres correspond to the most profitable movies as well as the movies with the best ROI?**
- I will use the 2010-2019 dataset for this analysis as it has the most complete genre information.

In [ ]:
#arranging the 2010 - 2019 movies by most profit and most ROI.
profit_df= recent_movies_2010.sort_values('total_profit', ascending=False)
roi_df = recent_movies_2010.sort_values('ROI', ascending=False)
top_profit = profit_df[0:500]
all_profit = profit_df[0:1250] #new variable as of 9/30
top_roi = roi_df[0:500]

In [ ]:
def genre_df(df): #function to get the value counts of the genre columns in a dataframe
    genre_df = df[['genre_1', 'genre_2' , 'genre_3']]
    genre_counts= genre_df.apply(pd.Series.value_counts)
    genre_df = genre_counts.fillna(0)
    genre_df['total'] = genre_counts.sum(axis=1)
    genre_df = genre_df.reset_index()
    return genre_df

In [ ]:
#movies that did not make a profit
loss_df= recent_movies_2010.sort_values('total_profit', ascending=True)
loss_df = loss_df[0:800]

In [ ]:
loss_genre_df = genre_df(loss_df)
loss_genre_df = loss_genre_df.reset_index()
loss_genre_df = loss_genre_df.drop(loss_genre_df[['level_0','genre_1', 'genre_2', 'genre_3']], axis=1)
loss_genre_df = loss_genre_df.sort_values('total', ascending=False)
loss_genre_df.head()

In [ ]:
#Genre DF for ALL movies that are in genre_df_2010 
genre_df_2010 = genre_df(recent_movies_2010)
genre_df_2010 = genre_df_2010.drop(genre_df_2010[['genre_1', 'genre_2', 'genre_3']], axis=1)
genre_df_2010 = genre_df_2010.sort_values('total', ascending = False)
genre_df_2010

In [ ]:
#These are the genres from the 500 most profitable movies from 2010-2019
genre_df_profit = genre_df(top_profit)
genre_df_profit = genre_df_profit.sort_values('total', ascending=False)
genre_df_profit = genre_df_profit.drop(genre_df_profit[['genre_1', 'genre_2', 'genre_3']], axis=1)

genre_df_profit

### Genre Success Rate

In [ ]:
#function for determiningmsuccess rate

def success_rate_df(df):
    success_rate = df.apply(lambda x: x['Profit'] / x['Total'], axis=1)
    df['success_rate'] = success_rate
    df['success_rate'] = df['success_rate'].map("{:.2%}".format)
    df = df.sort_values('success_rate', ascending=False)

    
    



In [ ]:
#Comparing genres of top 500 profitable movies from 2010 df with genres of movies that lost money.

genre_comparison_df = pd.merge(genre_df_profit, loss_genre_df, on='index', how='inner')
genre_comparison_df = genre_comparison_df.rename(columns={"total_x": "Profit", "total_y": "Loss"})
total_genre_count = genre_comparison_df.apply(lambda x: x['Profit'] + x['Loss'], axis=1)
genre_comparison_df['Total'] = total_genre_count
success_rate = genre_comparison_df.apply(lambda x: x['Profit'] / x['Total'], axis=1)
genre_comparison_df['success_rate'] = success_rate
genre_comparison_df['success_rate'] = genre_comparison_df['success_rate'].map("{:.2%}".format)
genre_comparison_df = genre_comparison_df.sort_values('success_rate', ascending=False)

genre_comparison_df

In [ ]:
#slicing success rate for top 500 movies
genre_success_df = genre_comparison_df
genre_success_df = genre_success_df.drop(genre_success_df[['Profit', 'Loss', 'Total']], axis=1)

In [ ]:
genre_success_df['top_500_success_rate'] = genre_success_df['success_rate']
genre_success_df.head(2)

In [ ]:
genre_success_df = genre_comparison_df
genre_success_df = genre_success_df.drop(genre_success_df[['Profit', 'Loss', 'Total']], axis=1)

In [ ]:
#These are the genres from ALL profitable movies from 2010-2019
genre_df_all_profit = genre_df(all_profit)
genre_df_all_profit = genre_df_all_profit.sort_values('total', ascending=False)
genre_df_all_profit = genre_df_all_profit.drop(genre_df_all_profit[['genre_1', 'genre_2', 'genre_3']], axis=1)

genre_df_all_profit.head(3)

In [ ]:
#Comparing genres of ALL profitable movies from 2010 df with genres of movies that lost money.

genre_comparison_df_2 = pd.merge(genre_df_all_profit, loss_genre_df, on='index', how='inner')
genre_comparison_df_2 = genre_comparison_df_2.rename(columns={"total_x": "Profit", "total_y": "Loss"})
total_genre_count_2 = genre_comparison_df_2.apply(lambda x: x['Profit'] + x['Loss'], axis=1)
genre_comparison_df_2['Total'] = total_genre_count_2

success_rate_df(genre_comparison_df_2)
genre_comparison_df_2 = genre_comparison_df_2.sort_values('success_rate',ascending=False)
genre_comparison_df_2

In [ ]:
genre_success_df_2 = genre_comparison_df_2
genre_success_df_2 = genre_success_df_2.drop(genre_success_df_2[['Profit', 'Loss', 'Total']], axis=1)
genre_success_df_2.head(2)

In [ ]:
genre_success_df_2['any_profit_success_rate'] = genre_success_df_2['success_rate']
genre_success_df_2

In [ ]:
genre_success_combined = pd.merge(genre_success_df, genre_success_df_2, on='index', how='inner')
genre_success_combined = genre_success_combined.drop(genre_success_combined[['success_rate_x', 'success_rate_y']], axis=1)
genre_success_combined.head(2)

### Genre Top Profit & ROI

In [ ]:
#These are the genres from the 500 movies with the best ROI from 2010-2019
genre_df_roi = genre_df(top_roi)
genre_df_roi = genre_df_roi.sort_values('total', ascending=False)
genre_df_roi = genre_df_roi.drop(genre_df_roi[['genre_1', 'genre_2', 'genre_3']], axis=1)
genre_df_roi.head(3)

In [ ]:
#resetting the indexes so that I can merge them
genre_df_roi = genre_df_roi.reset_index()
genre_df_profit = genre_df_profit.reset_index()


In [ ]:
genre_df_merged = pd.merge(genre_df_profit, genre_df_roi, on= 'index', how='left')
genre_df_merged = genre_df_merged.rename(columns={'total_x': 'profit_genres', 'total_y': 'roi_genres'})
genre_df_merged.sort_values('roi_genres', ascending=False)

In [ ]:
#drop the extra columns.
#genre_df_merged = genre_df_merged.drop(['genre_1_x', 'genre_2_x', 'genre_3_x',
                                         #'genre_1_y', 'genre_2_y', 'genre_3_y'], axis=1)

#re-label the two columns that I am keeping.
#genre_df_merged = genre_df_merged.rename(columns={'total_x': 'profit_genres', 'total_y': 'roi_genres'})

#get the total for each genre and sort from most to least.
#genre_df_merged['Total'] = genre_df_merged.apply(lambda x: x['profit_genres'] + x['roi_genres'], axis=1)
#genre_df_merged = genre_df_merged.sort_values('Total', ascending = False)
#genre_df_merged

### Genre Performance Dataframe

In [ ]:
genre_performance_df = pd.merge(genre_success_combined, genre_df_merged, on= 'index', how='left')
genre_performance_df['total_in_top_500_profit'] = genre_performance_df['profit_genres']
genre_performance_df['total_in_top_500_roi'] = genre_performance_df['roi_genres']
genre_performance_df = genre_performance_df.drop(genre_performance_df[['level_0', 'profit_genres', 'roi_genres']], axis=1)

genre_performance_df.head(2)

In [ ]:
#success rate of the top 500 most profitable movies from 2010-2019
genre_performance_success_1 = genre_performance_df.sort_values('top_500_success_rate', ascending=False)

#success rate of all profitable movies from 2010-2019
genre_performance_success_2 = genre_performance_df.sort_values('any_profit_success_rate', ascending=False)

# number of movies in the top 500 most profitable movies from 2010-2019
genre_performance_500_profit = genre_performance_df.sort_values('total_in_top_500_profit', ascending=False)

# number of movies in the top 500 ROI from 2010-2019
genre_performance_500_roi = genre_performance_df.sort_values('total_in_top_500_roi', ascending=False)

In [ ]:
genre_performance_500_profit

In [ ]:
#genre_performance_df['total_in_top_500_profit'] = genre_performance_df['profit_genres']
#genre_performance_df['total_in_top_500_roi'] = genre_performance_df['roi_genres']
#genre_performance_df

### Genre Analysis

**GENRE ANALYSIS GOES HERE**

#### Genre Visualization

In [ ]:
df =genre_df_merged

fig, ax = plt.subplots(figsize = (20,10))

ax.bar(x=df['index'], height=df['Total'])

ax.set_xlabel('Genre Types')
ax.set_ylabel('Number of Movies')
ax.set_title('Genre Types Among Top Movies');


In [ ]:
#attempting to show roi and profit
df =genre_df_merged

fig, ax = plt.subplots(figsize = (20,10))

ax.bar(x=df['index'], height=df['profit_genres'], color='gold')
ax.bar(x=df['index'], height=df['roi_genres'], color='purple')


ax.set_xlabel('Genre Types')
ax.set_ylabel('Number of Movies')
ax.set_title('Genre Types Among Top 500 Movies');


### Budget Analysis

**How much does Microsoft need to spend on these movies?**

In [ ]:
roi_budget_df = top_roi.sort_values('production_budget', ascending=True)
low_budget_roi = roi_budget_df[0:150] #the 150 lowest budget movies in the top 500 ROI

high_budget_roi = roi_budget_df.sort_values('production_budget', ascending=False)
high_budget_roi = high_budget_roi[0:150] #the 150 highest budget movies in the top 500 ROI




In [ ]:
low_budget_roi.head(150)

In [ ]:
low_budget_roi.describe()

In [ ]:
high_budget_roi.head(150)

In [ ]:
high_budget_roi.describe()

#### Genres by Budget

In [ ]:
#These are the genres for the 150 highest budget movies out of the top 500 ROI
high_budget_genres = genre_df(high_budget_roi)
high_budget_genres = high_budget_genres.reset_index()
high_budget_genres.sort_values('total', ascending=False)

In [ ]:
low_budget_roi.describe()

In [ ]:
#These are the genres for the 150 lowest budget movies out of the top 500 ROI
low_budget_genres = genre_df(low_budget_roi)
low_budget_genres.sort_values('total', ascending=False)

## Visualization Archive

In [ ]:
#GENRE TYPES AMONG MOVIES THAT LOST MONEY
df = loss_genre_df

fig, ax = plt.subplots(figsize = (20,10))

ax.bar(x=df['index'], height=df['total'], color='red')


ax.set_xlabel('Genre Types')
ax.set_ylabel('Number of Movies')
ax.set_title('Genre Types Among Movies that Lost Money');


In [ ]:
#Visualization for genre comparison dataframes. Doesn't work because i got rid of Difference column.

df = genre_comparison_df

fig, ax = plt.subplots(figsize = (20,10))

ax.bar(x=df['index'], height=df['difference'],  color=(df['difference'] >= .50).map({True: 'g',
                                                    False: 'r'}))


ax.set_xlabel('Genre Types', size=20)
ax.set_ylabel('Number of Movies', size=20)
ax.set_title('Genre Types Among Movies', size=20);


## ARCHIVE

In [ ]:
#difference = genre_comparison_df.apply(lambda x: x['Profit'] - x['Loss'], axis=1)
#genre_comparison_df['Difference'] = difference
#genre_comparison_df


In [ ]:

#low_budget = budget_df[0:100]
#low_budget

In [ ]:
#These are the genres for the 100 lowest budget movies out of the top 325 ROI
#low_budget_genres = genre_df(low_budget)
#low_budget_genres.sort_values('total', ascending=False)

In [ ]:
#high_budget_2010_roi = budget_df.sort_values('production_budget', ascending=False)
#high_budget_2010_roi = high_budget_2010_roi[0:100]


In [ ]:
#low_budget_2010 = recent_movies_2010.sort_values('production_budget', ascending=True)
#low_budget_2010.head(500)

In [ ]:
#attempting to get data on budgets related to genre
#test_df = recent_movies_2010.sort_values('genre_1', ascending=True)
#test_df.head(100)

In [ ]:
#action_movies = recent_movies_2010[recent_movies_2010['genre_1'] == 'Action']
#action_movies

In [ ]:
#horror_1 = recent_movies_2010[recent_movies_2010['genre_1'] == 'Horror']
#horror_2 = recent_movies_2010[recent_movies_2010['genre_2'] == 'Horror']
#horror_3 =  recent_movies_2010[recent_movies_2010['genre_3'] == 'Horror']
#horror_df = horror_1.append([[horror_2, horror_3]])                     

In [ ]:
#is there a coorelation between genre and performance?
#budget and genre?

In [ ]:
#merged_rt_df_2 = merged_rt_df

In [ ]:
#merged_rt_df['runtime'].isna().sum()

In [ ]:
#corrected_runtime = merged_rt_df_2['runtime'].fillna(0)

In [ ]:
#corrected_runtime

In [ ]:
#corrected_runtime_2 = corrected_runtime.str[:3]
#corrected_runtime_2

In [ ]:
#corrected_runtime_2.isna().sum()

In [ ]:
#corrected_runtime_3 = corrected_runtime_2.fillna(0)

In [ ]:
#corrected_runtime_3.value_counts()

In [ ]:
#merged_rt_df_2['runtime'] = corrected_runtime_3

In [ ]:
#merged_rt_df_2.info()

In [ ]:
#merged_rt_df_2 = merged_rt_df_2.drop('box_office', axis=1)
#merged_rt_df_2.head(10)

In [ ]:
#merged_rt_df_2['runtime'].isna().sum()



In [ ]:
#droping null values so I can convert the reamining ones to float.
#merged_rt_df_2['runtime'].dropna(NaN)

In [ ]:
#cleaned_runtime = merged_rt_df_2['runtime'].str[:3]
#cleaned_runtime.tail(5)

In [ ]:
#cleaned_theater_date = pd.to_datetime(merged_rt_df_2['theater_date'])
#cleaned_theater_date

In [ ]:
#merged_rt_df_2["theater_date"] = cleaned_theater_date
#merged_rt_df_2

In [ ]:
#df = merged_rt_df_2

In [ ]:
#duplicates = df[df.duplicated()]
#print(len(duplicates))
#duplicates.head()


In [ ]:
#rt_duplicates = merged_rt_df_2[[merged_rt_df_2['id']].duplicated]



In [ ]:
#merged_rt_df_2.info()

In [ ]:
#merged_df_2.info()

In [ ]:
#merged_rt_df['runtime'][299]

In [ ]:
merged_rt_df.sort_values('theater_date', ascending=False).head(100)

In [ ]:
merged_rt_df

In [ ]:
# Here you run your code to clean the data

## Data Analysis
Describe and justify the process for analyzing or modeling the data.

***
Questions to consider:
* How did you analyze or model the data?
* How did you iterate on your initial approach to make it better?
* Why are these choices appropriate given the data and the business problem?
***

###  What I did.

There were two main components of data that I was interested in. The first was Financial data, specificially, how much
did it cost to make the movie and how much money did it make at the box office. The second was information on what genre (or genres) the best performing movies had.

The Numbers dataset had the financial information that I needed. I was able to use the provided production budget and worldwide gross to calculate the total profit or loss for each movie. 

I then merged all the IMDB datasets into a master data set using the IMDB ID (tconst) as a primary key. There was a lot of extra data there that I didn't end up using, but it did provide the primary title and orginial title, which I could use to link to The Numbers.

### Limitations of the data:

Most of the genre data that I have is from 2010 - 2019. While I would have liked to have consistent genre data for all of the movies, this data is from the most relevant movies for answering the business problem. Microsoft wants data on what types of movies are performing well NOW, so this data was very valuable. I was still able to get financial data for movies from previous years, so it all worked to paint a good picture of the movie industry.

### Challenges along the way (and how I dealt with them):

The biggest challenge that I faced was merging the IMDB data with data from The Numbers. While both provided titles for each movie, there was a little descrepency with some more common titles. Most notably, Avatar, the highest grossing movie in the dataset, was being linked with a low budget horror movie of the same name. I was able to correctly merge the data by using both the primary title and original title from IMDB to determine which was correct. Also, dropping duplicates resolved a lot of the problems.

Another challenge was that the genre field on the IMDB dataset had up to 3 genres all contained together in one string, making it hard to search the data. I was able to create a function which seperated this string into 3 seperate genre fields, as well counting the occurances of each genre through value counts.

## Questions that I want to answer:
    
    1) What are the genres of the movies that made the most profit?
    2) What are the genres of the movies with the best ROI?
    3) How much money do you need to spend to make these movies?
    4) Maybe look at directors/studios/etc?

Findings from Analysis:



**Recommendations:**
    1. 

## Evaluation
Evaluate how well your work solves the stated business problem.

***
Questions to consider:
* How do you interpret the results?
* How well does your model fit your data? How much better is this than your baseline model?
* How confident are you that your results would generalize beyond the data you have?
* How confident are you that this model would benefit the business if put into use?
***

## Conclusions
Provide your conclusions about the work you've done, including any limitations or next steps.

***
Questions to consider:
* What would you recommend the business do as a result of this work?
* What are some reasons why your analysis might not fully solve the business problem?
* What else could you do in the future to improve this project?
***